06.bowling_object/game.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def initialize(scores)
+    @frames = [Frame.new(0)]
+    @special_frames = []
+    @score_board = scores.gsub('X', '10').split(',').map(&:to_i)
+
+    @score_board.each do |shot_pin|
+      record_shot(shot_pin)
+    end
+  end
+
+  def record_shot(pins)
+    calc_bonus(pins)
+    frame = @frames.last
+    frame.record_shot(pins)
+    return if frame.last_frame?
+
+    @frames << Frame.new(@frames.size) if frame.finished?
+    @special_frames << frame if frame.strike? || frame.spare?
+  end
+
+  def score
+    @frames.sum(&:score)
+  end
+
+  private
+
+  def calc_bonus(pins)
+    @special_frames.each do |special_frame|
+      special_frame.add_bonus(pins) if special_frame.need_bonus?
```

Comment:
```suggestion
      special_frame.add_bonus(pins)
```

special_frame 自身がbonusを追加すべきどうかを判断できそうです

Reply:
frame自身に判断させる方法を考えてみましたが、現在の方法より良いものは思いつきませんでした。
以下が考えた内容です

* ループの中で分岐を作り、ストライクだったら2個まで、スペアだったら1個までボーナスを加算する
    - やってることは結局今と同じでコードが増えるだけなので無し
* Frameクラスのインスタンス変数としてフラグを追加する
    - 今より複雑になるだけなので無し
    
以上から今の実装が一番分かりやすいかなーと考えています、他に良い方法がありそうでしたらもう一度考えてみます。